swarm/network/discovery_test.go

```diff
@@ -160,7 +160,10 @@ func testInitialPeersMsg(t *testing.T, peerPO, peerDepth int) {
 	// block until control peer is found among hive peers
 	found := false
 	for attempts := 0; attempts < 20; attempts++ {
-		if _, found = hive.peers[peerID]; found {
+		hive.lock.Lock()
```

Comment:
Yes, you are right. I did not want to introduce new method for this purpose. I think that it would be much cleaner to have it. Since you suggested it, I'll add it.